### PR TITLE
Remove powervs-utils package dependency

### LINKED ISSUE #643: Handle the region and zones mapping properly
/kind feature
/area provider/ibmcloud

**Describe the solution you'd like**
Currently https://github.com/ppc64le-cloud/powervs-utils is used to find the mapping, need to find a way to avoid using this. 

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by Prajyot-Parab**
Signed-off-by: Prajyot-Parab <prajyot.parab2@ibm.com>

**What this PR does / why we need it**:
- Remove powervs-utils package dependency

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #643 

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
Remove powervs-utils package dependency
```

### CODE REVIEW

File: pkg/endpoints/endpoints.go [hunk 1]
```diff
@@ -142,3 +143,18 @@ func FetchRCEndpoint(serviceEndpoint []ServiceEndpoint) string {
 	}
 	return ""
 }
+
+// CostructRegionFromZone Calculate region based on location/zone.
+func CostructRegionFromZone(zone string) string {
```
Comment (mkumatag) [inline pkg/endpoints/endpoints.go:148]: is this same logic been used in the powervs sdk?

File: pkg/endpoints/endpoints_test.go [hunk 2]
```diff
@@ -255,3 +255,34 @@ func TestFetchRCEndpoint(t *testing.T) {
...
 	}
 }
+
+func TestCostructRegionFromZone(t *testing.T) {
+	testCases := []struct {
+		name           string
+		zone           string
+		expectedRegion string
+	}{
+		{
```
Comment (mkumatag) [inline pkg/endpoints/endpoints_test.go:265]: can we add all the remaining regions for having better coverage.?

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: pkg/endpoints/endpoints.go (hunk 1 above)
Comment (Prajyot-Parab) [reply pkg/endpoints/endpoints.go:148]: Yes.

**Review by mkumatag — APPROVED** [review]
/lgtm

**Pull request merged**